### PR TITLE
HOT-2474-Refactor-Relationship-Dropdown

### PR DESCRIPTION
**Pull request by amanbop**
### Jira Story

HOT-2474 Refactor Relationship Dropdown
https://osi-cwds.atlassian.net/browse/HOT-2474
## Description
This is a refactor of the Intelligent Relationship Dropdown functionality incorporated into Edit Relationships form two sprints ago. This refactor is just to make code more manageable. It will/should NOT introduce ANY change to the existing functionality of the dropdown.

## Tests
- [x] I have included unit tests 
- [x] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

All tests were included already in the last version of the relationship dropdown. All the tests are passing with new code. 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [X] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have ran all tests for the project.
- [X] I have ran lint/rubocop check for the changed/new files.
- [X] My code is in a stable state ready to be deployable, but not necessarily complete.
- [X] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by amanbop** [comment]
retest this please

File: app/javascript/common/relationship/relationship_dropdown/relationshipDropdown.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import {RELATIONSHIP_TYPES} from 'enums/RelationshipTypes'
 import moment from 'moment'
 
-const MALE_CODES = ['M', 'm', 'U', 'I']
-const FEMALE_CODES = ['F', 'f', 'U', 'I']
+const MALE_CODES = ['M', 'm', 'U']
```
Comment (karthik-sista) [inline app/javascript/common/relationship/relationship_dropdown/relationshipDropdown.js]: Curious - Why are we removing **Intersex (I)** here?

Comment (karthik-sista) [reply app/javascript/common/relationship/relationship_dropdown/relationshipDropdown.js]: i did not get **not in list of values**. so was curious.

Comment (amanbop) [reply app/javascript/common/relationship/relationship_dropdown/relationshipDropdown.js]: there is no 'I' in the enum/RelationshiTypes. The values are M,F,U,m,f only. U is treated same as I

**Pull request merged**